src/components/GoalPage/GoalPage.tsx

```diff
@@ -146,116 +137,76 @@ export const GoalPage = ({ user, ssrTime, params: { id } }: ExternalPageProps<{
         setCurrentProjectCache(null);
     });
 
-    const { toggleGoalWatching, toggleGoalStar } = useGoalResource(goal?.id, goal?._shortId);
-
-    const invalidateFn = useCallback(() => {
-        return utils.goal.getById.invalidate(id);
-    }, [id, utils.goal.getById]);
-
-    const stateMutation = trpc.goal.switchState.useMutation();
-    const onGoalStateChange = useCallback(
-        async (nextState: GoalStateChangeSchema['state']) => {
-            if (goal) {
-                await stateMutation.mutateAsync({
-                    state: nextState,
-                    id: goal.id,
-                });
-                invalidateFn();
-            }
+    const {
+        onChangeGoalProject,
```

Comment:
This name is not consistent with others and incorrect. `onGoalProjectChange` would be better